Projects/Presentation/Presentation/Sources/CoreMember/QrCode/QRScannerManger/QrsccannerRepresentable.swift

```diff
@@ -0,0 +1,104 @@
+//
+//  QrsccannerRepresentable.swift
+//  Presentation
+//
+//  Created by Wonji Suh  on 4/6/25.
+//
+
+import SwiftUI
+import VisionKit
+
+import LogMacro
+
+struct QrsccannerRepresentable: UIViewControllerRepresentable {
+  @Binding var shouldStartScanning: Bool
+  @Binding var scannedText: String
+  
+  var dataToScanFor: Set<DataScannerViewController.RecognizedDataType>
+  
+  // MARK: - Coordinator
+  class Coordinator: NSObject, DataScannerViewControllerDelegate {
+    var parent: QrsccannerRepresentable
+    
+    init(_ parent: QrsccannerRepresentable) {
+      self.parent = parent
+    }
+    
+    // 사용자가 탭했을 때 호출 (원한다면 사용)
+    func dataScanner(_ dataScanner: DataScannerViewController, didTapOn item: RecognizedItem) {
+      switch item {
+      case .text(let text):
+        parent.scannedText = text.transcript
+      case .barcode(let barcode):
+        parent.scannedText = barcode.payloadStringValue ?? "Unable to decode the scanned code"
+      default:
+        #logDebug("Unexpected item")
+      }
+    }
+    
+    // 인식된 아이템이 업데이트될 때 호출됨
+    func dataScanner(_ dataScanner: DataScannerViewController,
+                     didUpdate items: [RecognizedItem],
+                     allItems: [RecognizedItem]) {
+      #logDebug("인식된 아이템: \(items)")
+      for item in items {
+        if case let .barcode(barcode) = item {
+          #logDebug("QR 코드 인식됨: \(barcode.payloadStringValue ?? "nil")")
+          parent.scannedText = barcode.payloadStringValue ?? "Unable to decode the scanned code"
+          // 비동기 방식으로 스캔 중지
+          DispatchQueue.main.async {
+            dataScanner.stopScanning()
+          }
+          break
+        }
+      }
+    }
+  }
+  
+  // MARK: - UIViewControllerRepresentable
+  func makeUIViewController(context: Context) -> DataScannerViewController {
+    let dataScannerVC = DataScannerViewController(
+      recognizedDataTypes: dataToScanFor,
+      qualityLevel: .accurate,
+      recognizesMultipleItems: true,
+      isHighFrameRateTrackingEnabled: true,
+      isPinchToZoomEnabled: true,
+      isGuidanceEnabled: true,
+      isHighlightingEnabled: true
+    )
+    dataScannerVC.delegate = context.coordinator
+    return dataScannerVC
+  }
+  
+  func updateUIViewController(_ uiViewController: DataScannerViewController, context: Context) {
+    if shouldStartScanning {
+      Task {
+        do {
+          try await startScanningAsync(uiViewController)
+        } catch {
+          #logDebug("Failed to start scanning: \(error)")
+        }
+      }
+    } else {
+      uiViewController.stopScanning()
+    }
```

Comment:
```suggestion
    let isScanning = uiViewController.isScanning
    if shouldStartScanning, !isScanning {
      do {
        try uiViewController.startScanning()
      } catch {
        #logDebug("Failed to start scanning: \(error)")
      }
    } else if !shouldStartScanning, isScanning {
      uiViewController.stopScanning()
    }
```
이렇게 수정하는건 어떨까요?

기존 코드는 updateUIViewController가 실행될 때마다 `startScanningAsync` 가 실행되어서 상태 변화가 일어나면 오동작이 일어날 수 있어보여서 isScanning 을 통해서 예외처리를 해주는 게 좀 더 안전할 것 같아요!

updateUIViewController도 무조건 메인스레드에서 동작되는게 보장되고, DataScannerViewController의 startScanning()가 MainActor에서 실행되기 때문에 따로 async/await으로 만들지 않고 사용해도 문제없을 것 같습니다!

Reply:
@honghoker  역시 은표님  감사합니다!